engine/class_modules/priest/sc_priest_holy.cpp

```diff
@@ -245,23 +245,38 @@ void priest_t::generate_apl_holy_d()
     // Potions
     default_list->add_action("potion,if=buff.bloodlust.react|(raid_event.adds.up&(raid_event.adds.remains>20|raid_event.adds.duration<20))|target.time_to_die<=30");
 
+
+    // Racials
+    if (race != RACE_BLOOD_ELF)
+    {
+        for (const std::string& racial_action : get_racial_actions())
+        {
+            default_list->add_action(racial_action);
+        }
+    }
+// Belf is a DPS decrease to use until like 20 targets for now
+//  if (race == RACE_BLOOD_ELF)
+//   {
+//           default_list->add_action("arcane_torrent,if=mana.pct<=95");
+//   }
+
     // Default APL	
     default_list->add_action(this, "Holy Fire", "if=dot.holy_fire.ticking&(dot.holy_fire.remains<=gcd|dot.holy_fire.stack<2)&spell_targets.holy_nova<7");
     default_list->add_action(this, "Holy Word: Chastise", "if=spell_targets.holy_nova<5");
     default_list->add_action(this, "Holy Fire", "if=dot.holy_fire.ticking&(dot.holy_fire.refreshable|dot.holy_fire.stack<2)&spell_targets.holy_nova<7");
     default_list->add_action("berserking,if=raid_event.adds.in>30|raid_event.adds.remains>8|raid_event.adds.duration<8");
     default_list->add_action("fireblood,if=raid_event.adds.in>20|raid_event.adds.remains>6|raid_event.adds.duration<6");
     default_list->add_action("ancestral_call,if=raid_event.adds.in>20|raid_event.adds.remains>10|raid_event.adds.duration<10");
-    default_list->add_talent(this, "Divine Star", "if=(raid_event.adds.in>5|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.divine_star>1" );
-    default_list->add_talent(this, "Halo", "if=(raid_event.adds.in>14|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.halo>0" );
+    default_list->add_talent(this, "Divine Star", "if=(raid_event.adds.in>5|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.divine_star>1");
+    default_list->add_talent(this, "Halo", "if=(raid_event.adds.in>14|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.halo>0");
```

Comment:
same thing as above with speed of APL `spell_targets.halo>0` moving to the front